python/ray/tune/schedulers/hyperband.py

```diff
@@ -254,20 +253,23 @@ def _process_bracket(
             # kill bad trials
             self._num_stopped += len(bad)
             for t in bad:
+                # We should have already cleaned up an errored/terminated trial
+                assert t.status not in (Trial.ERROR, Trial.TERMINATED), (
+                    f"Bad trial {t.trial_id} is in an invalid state: {t.status}\n"
+                    "Expected trial to be either PAUSED, PENDING, or RUNNING.\n"
+                    "If you encounter this, please file an issue on the Ray Github."
+                )
+
                 if t.status == Trial.PAUSED:
                     trial_runner.stop_trial(t)
-                elif t.status == Trial.RUNNING:
+                elif t.status in Trial.RUNNING:
```

Comment:
I initially did this, but it started giving some errors on the execution side.

```
  File "/Users/justin/Developer/justinvyu-dev/python/ray/tune/execution/trial_runner.py", line 748, in _process_trial_results
    decision = self._process_trial_result(trial, result)
  File "/Users/justin/Developer/justinvyu-dev/python/ray/tune/execution/trial_runner.py", line 791, in _process_trial_result
    decision = self._scheduler_alg.on_trial_result(
  File "/Users/justin/Developer/justinvyu-dev/python/ray/tune/schedulers/hb_bohb.py", line 88, in on_trial_result
    bracket.update_trial_stats(trial, result)
  File "/Users/justin/Developer/justinvyu-dev/python/ray/tune/schedulers/hyperband.py", line 470, in update_trial_stats
    assert trial in self._live_trials
AssertionError
```

Basically, the trial is pending, gets cleaned up in the scheduler state, then starts running and returns a result. Then, the scheduler is confused because it already cleaned up the trial.

Basically, we should be calling `stop_trial` for bad trials, but that didn't really work when I tried it -- the example started to hang after a certain number of trials (see comment above).